### PR TITLE
CI: Add Windows wheels for the free-threaded build

### PR DESCRIPTION
**Pull request by lysnikolaou**
- [x] Related to #59057
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -160,7 +160,13 @@ free-threaded-support = true
 before-build = "PACKAGE_DIR={package} bash {package}/scripts/cibw_before_build.sh"
 
 [tool.cibuildwheel.windows]
-before-build = "pip install delvewheel && bash {package}/scripts/cibw_before_build.sh"
+before-build = "pip install delvewheel && bash {package}/scripts/cibw_before_build_windows.sh"
+before-test = "bash {package}/scripts/cibw_before_test_windows.sh"
+test-command = """
+  set PANDAS_CI='1' && \
```
Comment (mroeschke) [inline pyproject.toml:166]: Could the `PANDAS_CI` be set in the `before-test` so that this section can re-use the `test-command`?

Comment (lysnikolaou) [reply pyproject.toml:166]: We could do that, yes, but all `cibuildwheel` custom configs include a different `test-command`. The Windows one has a slight difference as well.

**Comment by lysnikolaou** [comment]
Can we merge this? Is there anything else I'd need to do before we can do it?

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @lysnikolaou

**Pull request merged**